Add setting for keep-alive duration for oidc back-channel (#87773)
elastic/elasticsearch: docs/reference/settings/security-settings.asciidoc

Defaults to `200`.
// end::oidc-http-max-endpoint-connections-tag[]

// tag::oidc-http-connection-pool-ttl-tag[]
`http.connection_pool_ttl` {ess-icon}::
(<<static-cluster-setting,Static>>)
Controls the behavior of the http client used for back-channel communication to
the OpenID Connect Provider endpoints. Specifies the time-to-live of connections
in the connection pool (default to 3 minutes). A connection is closed if it is
idle for more than the specified timeout.

The server can also set the `Keep-Alive` HTTP response header. The effective
time-to-live value is the smaller value between this setting and the `Keep-Alive`

> reponse header. Configure this setting to `-1` to let the server dictates the value.

Reviewer: ```suggestion
reponse header. Configure this setting to `-1` to let the server dictate the value.
```

response header. If the header is not set by the server, the time-to-live is infinite meaning

> that connections never expire.

Reviewer: I think we should reword this, but I'll come back with a suggestion later.

Author: I reword'd this paragraph and also made it consistent to the new behaviour of handling keep-alive response header.

> the time-to-live is infinite meaning that connections never expire.

Reviewer: ```suggestion
the time-to-live is infinite and connections never expire.
```
